### PR TITLE
OTT-784: Added NonBidStatusCodes file

### PR DESCRIPTION
**Pull request by PubMatic-OpenWrap**

### CODE REVIEW

File: openrtb3/no_bid_status_codes.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	NoBidDailyUserCapMet                    NonBidStatusCode = 9   // No Bid - Daily User Cap Met
+	NoBidDailyDomainCapMet                  NonBidStatusCode = 10  // No Bid - Daily Domain Cap Met
+	NoBidAdstxtAuthorizationUnavailable     NonBidStatusCode = 11  // No Bid - Ads.txt Authorization Unavailable
+	NoBidAdstxtAuthorizationViolation       NonBidStatusCode = 12  // No Bid - Ads.txt Authorization Violation
+	NoBidAdscertAuthenticationUnavailable   NonBidStatusCode = 13  // No Bid - Ads.cert Authentication Unavailable
+	NoBidAdscertAuthenticationViolation     NonBidStatusCode = 14  // No Bid - Ads.cert Authentication Violation
+	NoBidInsufficientAuctionTime            NonBidStatusCode = 15  // No Bid - Insufficient Auction Time
+	NoBidIncompleteSupplychain              NonBidStatusCode = 16  // No Bid - Incomplete SupplyChain
+	NoBidBlockedSupplyChainNodeObj          NonBidStatusCode = 17  // No Bid - Blocked SupplyChain Node
+	NoBidGeneralError                       NonBidStatusCode = 100 // Error - General
```
Comment (ashishshinde-pubm) [inline openrtb3/no_bid_status_codes.go]: Any difference between "NoBidGeneralError" and "NoBidGeneral"

File: openrtb3/no_bid_status_codes.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package openrtb3
```
Comment (ShriprasadM) [inline openrtb3/no_bid_status_codes.go]: rename file to to `non_bid_status_code.go`

File: openrtb3/no_bid_status_codes.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// NonBidStatusCode represents OpenRTB Loss Reason Code enumeration.
+//
+// It lists the options for an exchange to inform a bidder as to the reason why they did not win an item.
+type NonBidStatusCode int64
+
+// NonBidStatusCode options.
+//
+// Values 500+ are exchange specific values; should be communicated with buyers beforehand.
+const (
+	NoBidWinningBid                         NonBidStatusCode = -1  // Winning Bid
```
Comment (ShriprasadM) [inline openrtb3/no_bid_status_codes.go]: Just worrying around  `NonBidStatusCode`. Whether it should be part of the specification. It's very internal to PubMatic's Analaytics. 
Let us discuss around it with @pm-viral-vala , you and me

File: openrtb3/no_bid_status_codes.go (hunk 1 above)
Comment (ShriprasadM) [reply openrtb3/no_bid_status_codes.go]: @ashishshinde-pubm  as per the Prebid Proposal
`NoBidGeneral ` (`0`)  => Auction ran successfully without demand for the impression.
`NoBidGeneralError` (`100`) => Technical problem occurred during the auction.
Even I am not very much clear here on this interpretation. We can ask this to Prebid

**Review by ShriprasadM — CHANGES_REQUESTED** [review]
Please check review comments

**Pull request merged**